File: Automovil.py
class Automovil():

    def __init__(self, modelo, color, altura) -> None:
        self.modelo = modelo
        self.color = color
        self.altura = altura
        self.esta_encendido = False
        print(f"Auto '{modelo}' creado.")

    def encender(self):
        if self.esta_encendido:
            print("El auto ya está encendido.")
        else:
            print("Verificando si existe combustible...")
            print("Verificando controles...")
            print("El auto se ha encendido.")
            self.esta_encendido = True

    def acelerar(self):
        if self.esta_encendido:
            print("El auto está acelerando...")

    def frenar(self):
        if self.esta_encendido:
            print("El auto está frenando...")


auto1 = Automovil("Lexus", "Azul", 1.95)
auto1.encender()
auto1.acelerar()
print(auto1.color)

print("=====================================")

auto2 = Automovil("Omega", "Rojo", 2.10)
auto2.encender()
auto2.acelerar()

print("=====================================")

auto3 = Automovil("Celta", "Verde", 1.70)

File: Java/Demo_POO/src/Program/Automovil.java
package Program;

/**
 *
 * @author UskoKruM2010 (https://uskokrum2010.com)
 */
public class Automovil {

    public String modelo;
    public String color;
    public double altura;
    public boolean estaEncendido;

    public Automovil(String modelo, String color, double altura) {
        this.modelo = modelo;
        this.color = color;
        this.altura = altura;
        this.estaEncendido = false;
        System.out.println("Auto " + this.modelo + " creado.");
    }

    public void encender() {
        if (this.estaEncendido) {
            System.out.println("El auto ya está encendido.");
        } else {
            System.out.println("Verificando si existe combustible...");
            System.out.println("Verificando controles...");
            System.out.println("El auto se ha encendido.");
            this.estaEncendido = true;
        }
    }

    public void acelerar() {
        if (this.estaEncendido) {
            System.out.println("El auto está acelerando...");
        }
    }

    public void frenar() {
        if (this.estaEncendido) {
            System.out.println("El auto está frenando...");
        }
    }

}
